apis/core/lib/domain/queries/resources-types.ts

```diff
@@ -1,16 +1,56 @@
 import { Term, Quad } from 'rdf-js'
 import { CONSTRUCT } from '@tpluscode/sparql-builder'
 import { rdf } from '@tpluscode/rdf-ns-builders'
+import $rdf from 'rdf-ext'
+import rdfFetch from '@rdfjs/fetch'
+import env from '@cube-creator/core/env'
+import TermSet from '@rdfjs/term-set'
+import { log } from '../../log'
 import { parsingClient } from '../../query-client'
 
-export async function loadResourcesTypes(ids: Term[], client = parsingClient): Promise<Quad[]> {
-  return CONSTRUCT`?resource ${rdf.type} ?type`
+interface Clients {
+  sparql?: typeof parsingClient
+  fetch?: typeof rdfFetch
+}
+
+async function loadRemoteResourceTypes(id: Term, fetch: typeof rdfFetch): Promise<Quad[]> {
+  try {
+    const resource = await fetch(id.value, { factory: $rdf })
+    const dataset = await resource.dataset()
+    return dataset.match(id, rdf.type).toArray()
+  } catch (e) {
+    log(e)
+    return []
+  }
+}
+
+export async function loadResourcesTypes(ids: Term[], { sparql = parsingClient, fetch = rdfFetch }: Clients = {}): Promise<Quad[]> {
+  const { local, remote } = ids.reduce((separated, id) => {
+    if (id.value.startsWith(env.API_CORE_BASE)) {
+      separated.local.add(id)
+    } else {
+      separated.remote.push(loadRemoteResourceTypes(id, fetch))
+    }
+
+    return separated
+  }, {
+    local: new TermSet(),
+    remote: [] as Promise<Quad[]>[],
+  })
+
+  const typesFoundRemotely = Promise.all(remote)
+  const typesFoundLocally = CONSTRUCT`?resource ${rdf.type} ?type`
     .WHERE`
       values ?resource {
-        ${ids}
+        ${[...local]}
       }
     `
     .WHERE`GRAPH ?g {
         ?resource a ?type .
-    }`.execute(client.query)
+    }`.execute(sparql.query)
+
+  return [
+    ...await typesFoundLocally,
+    ...(await typesFoundRemotely).flatMap(arr => arr),
+  ]
```

Comment:
When a property with `sh:class` has a remote resource, that resource will be feted directly rather than looked up in our database